tests/explicit-nulls/neg/nullnull.scala

```diff
@@ -0,0 +1,9 @@
+// Test that Null | Null will not cause crash during typing.
+
+class Foo {
+  def foo1: Unit = {
+    val x: Null | Null | Null = ???
```

Comment:
Could you please add another test where `x` has type "String | JavaNull | Null", where we remember that x was a `JavaNull`?